subsystem/cluster_test.go

```diff
@@ -1519,6 +1519,70 @@ var _ = Describe("cluster install", func() {
 		})
 	})
 
+	Context("wait until all hosts are done", func() {
+		var (
+			clusterID         strfmt.UUID
+			misbehavingHostID strfmt.UUID
+		)
+		BeforeEach(func() {
+			clusterID = *cluster.ID
+			registerHostsAndSetRoles(clusterID, *infraEnvID, 6, cluster.Name, cluster.BaseDNSDomain)
+			cluster = getCluster(clusterID)
+			for _, h := range cluster.Hosts {
+				if h.Role == models.HostRoleWorker {
+					misbehavingHostID = *h.ID
+					break
+				}
+			}
+		})
+		stages := []models.HostStage{models.HostStageFailed, models.HostStageDone}
+		for i := range stages {
+			stage := stages[i]
+			It(fmt.Sprintf("full flow with single host in stage %s", string(stage)), func() {
+				By("installing cluster", func() {
+					installCluster(clusterID)
+					cluster = getCluster(clusterID)
+				})
+
+				By("move cluster to finalizing", func() {
+					for _, h := range cluster.Hosts {
+						if *h.ID != misbehavingHostID {
+							updateProgress(*h.ID, h.InfraEnvID, models.HostStageDone)
+						} else {
+							updateProgress(*h.ID, h.InfraEnvID, models.HostStageRebooting)
+						}
+					}
+					waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout,
+						IgnoreStateInfo)
+				})
+
+				By("complete installation. state should be still finalizing", func() {
+					completeInstallation(agentBMClient, clusterID)
+					waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout,
+						IgnoreStateInfo)
+				})
+
+				By("register host. move to pending-user-action", func() {
```

Comment:
based on [Bugzill#2107617 comment 3
](https://bugzilla.redhat.com/show_bug.cgi?id=2107617#c3), Can we assert here that the cluster installation progress did not decrease?

Reply:
OK